x-pack/legacy/plugins/reporting/export_types/png/server/execute_job/index.ts

```diff
@@ -26,12 +21,13 @@ import { generatePngObservableFactory } from '../lib/generate_png';
 
 type QueuedPngExecutorFactory = ExecuteJobFactory<ESQueueWorkerExecuteFn<JobDocPayloadPNG>>;
 
-export const executeJobFactory: QueuedPngExecutorFactory = function executeJobFactoryFn(
+export const executeJobFactory: QueuedPngExecutorFactory = async function executeJobFactoryFn(
+  reporting: ReportingPlugin,
   server: ServerFacade,
   elasticsearch: ElasticsearchServiceSetup,
-  parentLogger: Logger,
-  { browserDriverFactory }: { browserDriverFactory: HeadlessChromiumDriverFactory }
+  parentLogger: Logger
 ) {
+  const browserDriverFactory = await reporting.getBrowserDriverFactory();
```

Comment:
We're also using other accessor methods to get internal dependencies that are available after `setup`, like this `getBrowserDriverFactory`  and even `getEsqueue`